### DISCUSSION TITLE
Correct pattern for handling write protected objects

### DISCUSSION THREAD

**Discussion by tomfox747** [supabase/supabase#23379] [Questions] [answered]
Hi.
I'm trying to implement a profile picture image system in which anyone can view the image (public bucket) but only the authenticated owner of the profile image who has the image path stored in their profile row, can update or delete it.

Struggling to find the correct pattern or approach for this.
Any suggestions or help are welcome.
Thanks.

**Comment by GaryAustin1** [comment, marked as answer]
You would set up your storage policy for insert/select/update to have owner = auth.uid() as part of policy.  This assumes the user uploaded the avatar so the owner id would be theirs.

**Reply by tomfox747** [reply]
Thank you for pointing me in the direction, RLS is a new concept for me. Very impressed by Supabase overall.
